sdk/src/Services/DynamoDBv2/Custom/DocumentModel/DocumentBatchGet.cs

```diff
@@ -427,15 +427,22 @@ private Results GetAttributeItems()
             return results;
         }
 
-#if NETSTANDARD
-        private static void CallUntilCompletion(AmazonDynamoDBClient client, BatchGetItemRequest request, Results allResults)
-#else
         private static void CallUntilCompletion(IAmazonDynamoDB client, BatchGetItemRequest request, Results allResults)
-#endif
         {
             do
             {
-                var serviceResponse = client.BatchGetItem(request);
+#if AWS_ASYNC_API
+                // Cast the IAmazonDynamoDB to the concrete client instead, so we can access the internal sync-over-async methods
+                var internalClient = client as AmazonDynamoDBClient;
```

Comment:
nit: You can move the casting logic outside of the loop and only do it once.

Reply:
Fixed in https://github.com/aws/aws-sdk-net/pull/3388/commits/aa1e3273fae0ba8826c15a91902728713e06a1d4